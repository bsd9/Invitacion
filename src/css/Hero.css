@import url('https://fonts.googleapis.com/css2?family=Dancing+Script:wght@400;700&display=swap');

.hero {
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  padding: 20px;
  margin-left: 10vw;
  margin-right: 10vw;
}

.hero img {
  width: 80%;
  border-radius: 10px;
}

.hero .date {
  font-weight: 300;
  margin-top: 20px;
}

.hero h1 {
  font-family: 'Dancing Script', cursive;
  font-style: italic;
  font-size: 2.5rem;
}