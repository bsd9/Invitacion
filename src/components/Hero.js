import React from "react";
import "../css/Hero.css";

const Hero = ({ fecha = "26.04.2025", nombre = "María Fernanda\nSalazar Zuluaga", mensaje }) => {
  return (
    <div className="hero" data-testid="hero-container">
      <p className="date" data-testid="hero-fecha">
        _________________{fecha}_________________
      </p>
      <br />
      <h1 data-testid="hero-nombre">
        {nombre.split('\n').map((line, i) => (
          <React.Fragment key={i}>
            {line}
            {i < nombre.split('\n').length - 1 && <br />}
          </React.Fragment>
        ))}
      </h1>
      <br />
      <p data-testid="hero-mensaje">
        {mensaje || `Hoy celebro no solo mis 15 años, sino también cada momento que la vida me ha regalado. 
        Gracias por las experiencias que me han formado, por los aprendizajes, las sonrisas y hasta por los desafíos que me han enseñado a ser más fuerte. 
        Estoy agradecida por todo lo vivido y emocionada por lo que aún está por venir. ¡Gracias, vida, por cada regalo que me has dado!`}
      </p>
    </div>
  );
};

export default Hero;