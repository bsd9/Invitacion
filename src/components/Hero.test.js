import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Hero from './Hero';

describe('Componente Hero', () => {
  const propsBasicos = {
    fecha: "26.04.2025",
    nombre: "María Fernanda\nSalazar Zuluaga",
    mensaje: "Mensaje de prueba para el hero"
  };

  test('renderiza correctamente con props', () => {
    render(<Hero {...propsBasicos} />);
    
    const fechaElement = screen.getByTestId('hero-fecha');
    expect(fechaElement.textContent).toContain(propsBasicos.fecha);
    
    const nombreElement = screen.getByTestId('hero-nombre');
    expect(nombreElement.textContent).toMatch(/María Fernanda[\s\S]*Salazar Zuluaga/);
    
    expect(screen.getByTestId('hero-mensaje')).toHaveTextContent(propsBasicos.mensaje);
  });

  test('usa valores por defecto cuando no se proporcionan props', () => {
    render(<Hero />);
    
    const fechaElement = screen.getByTestId('hero-fecha');
    expect(fechaElement.textContent).toContain("26.04.2025");
    
    const nombreElement = screen.getByTestId('hero-nombre');
    expect(nombreElement.textContent).toMatch(/María Fernanda[\s\S]*Salazar Zuluaga/);
    
    expect(screen.getByTestId('hero-mensaje')).toBeInTheDocument();
  });

  test('tiene las clases CSS correctas', () => {
    const { container } = render(<Hero {...propsBasicos} />);
    expect(container.querySelector('.hero')).toBeInTheDocument();
    expect(container.querySelector('.date')).toBeInTheDocument();
  });

  test('maneja correctamente el salto de línea en el nombre', () => {
    render(<Hero nombre="Nombre\nApellido" />);
    const nombreElement = screen.getByTestId('hero-nombre');
    expect(nombreElement.textContent).toMatch(/Nombre[\s\S]*Apellido/);
  });

  test('coincide con el snapshot', () => {
    const { asFragment } = render(<Hero {...propsBasicos} />);
    expect(asFragment()).toMatchSnapshot();
  });
});