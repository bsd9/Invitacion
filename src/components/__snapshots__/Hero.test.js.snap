// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Hero coincide con el snapshot 1`] = `
<DocumentFragment>
  <div
    class="hero"
    data-testid="hero-container"
  >
    <p
      class="date"
      data-testid="hero-fecha"
    >
      _________________26.04.2025_________________
    </p>
    <br />
    <h1
      data-testid="hero-nombre"
    >
      Mar√≠a Fernanda
      <br />
      Salazar Zuluaga
    </h1>
    <br />
    <p
      data-testid="hero-mensaje"
    >
      Mensaje de prueba para el hero
    </p>
  </div>
</DocumentFragment>
`;
